packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -494,7 +494,10 @@ const PopoverPortal = ({
   const portalTo = usePortalTo(portal);
 
   return (
-    <FloatingPortal root={portalTo}>
+    <FloatingPortal
+      key={portalTo?.id} // workaround to force remount when `root` changes (see #2093)
```

Comment:
I think custom `portalContainer` should not be affected at all (it doesn't matter if there is an `id`) because it directly goes from `undefined` → `<div>` (rather than `undefined` → `<div id="1">` → `<div id="2">`).

Reply:
Normally, the internal portalContainer starts out as undefined, then resolves to the `<div id="1">` in the original window, then finally resolves to the `<div id="2">` in the new window.

When a custom `portalContainer` prop is used, it doesn't have that intermediate state.